### PR TITLE
Add Scala.js and drop Scala 2.10

### PR DESCRIPTION
**Pull request by takayahilton**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -39,18 +38,25 @@ lazy val chronoscala = (project in file("."))
...
   )
+  .settings(
+    libraryDependencies ++= Seq(
+      "org.scalacheck" %%% "scalacheck" % "1.14.3" % Test,
+      "org.scalatest" %%% "scalatest" % "3.1.1" % Test
+    )
+  )
+  .jsSettings(
+    libraryDependencies ++= Seq(
+      "io.github.cquiroz" %%% "scala-java-time" % "2.0.0-RC4",
```
Comment (ocadaruma) [inline build.sbt]: I feel we should wait until GA version has been released

Comment (ocadaruma) [reply build.sbt]: from scala-java-time gitter. seems RC4 (it planned to be GA) has some issue, and RC5 has been released soon after that.

image

File: build.sbt [hunk 2]
```diff
@@ -23,9 +22,9 @@ lazy val chronoscala = (project in file("."))
 
     publishMavenStyle := true,
 
-    scalaVersion := scala210,
+    scalaVersion := "2.11.12",
 
-    crossScalaVersions := Seq(scala210, "2.11.12", "2.12.10", "2.13.1"),
+    crossScalaVersions := Seq("2.11.12", "2.12.10", "2.13.1"),
```
Comment (sh0hei) [inline build.sbt:27]: The current latest version is as follows (2020/11/24)
cf. https://www.scala-lang.org/download/
- `2.11.12`
- `2.12.12`
- `2.13.4`

File: build.sbt [hunk 3]
```diff
@@ -39,18 +38,25 @@ lazy val chronoscala = (project in file("."))
...
     TaskKey[Unit]("checkScalariform") := {
       val diff = sys.process.Process("git diff").!!
       if (diff.nonEmpty) {
         sys.error("Working directory is dirty!\n" + diff)
       }
     }
   )
+  .settings(
+    libraryDependencies ++= Seq(
+      "org.scalacheck" %%% "scalacheck" % "1.14.3" % Test,
```
Comment (sh0hei) [inline build.sbt:50]: `1.15.1` Released.
cf. https://github.com/typelevel/scalacheck/releases

File: build.sbt [hunk 4]
```diff
@@ -39,18 +38,25 @@ lazy val chronoscala = (project in file("."))
...
+      "org.scalatest" %%% "scalatest" % "3.1.1" % Test
```
Comment (sh0hei) [inline build.sbt:51]: `3.2.3` Released.
cf. https://github.com/scalatest/scalatest/releases

File: build.sbt (hunk 1 above)
Comment (sh0hei) [reply build.sbt:56]: `2.0.0` Released 🎉 
cf. https://github.com/cquiroz/scala-java-time/releases/tag/v2.0.0

File: shared/src/test/scala/jp/ne/opt/chronoscala/Gens.scala [hunk 5]
```diff
@@ -40,3 +42,7 @@ trait Gens {
     end <- instantGen
   } yield Duration.between(start, end)
 }
+
+object Gens {
```
Comment (sh0hei) [inline shared/src/test/scala/jp/ne/opt/chronoscala/Gens.scala:46]: 👍

**Review by sh0hei — CHANGES_REQUESTED** [review]
@takayahilton I commented a few points 🙏

**Review by ocadaruma — APPROVED** [review]
LGTM, Thanks for your great contribution!

**Review by sh0hei — APPROVED** [review]
Thanks for your contributing @takayahilton !

**Pull request merged**